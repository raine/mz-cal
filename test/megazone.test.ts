import { back as nockBack } from 'nock'
import { DateTime } from 'luxon'
import { sortBy } from 'ramda'
import fs from 'fs'
import {
  parseCalendarHtml,
  parseCreateBookingHref,
  getEventsForWeek
} from '../src/megazone'

nockBack.setMode('lockdown')
nockBack.fixtures = __dirname + '/fixtures'

describe('getEventsForWeek', () => {
  it('returns events with player counts', async () => {
    const { nockDone } = await nockBack('calendar-nock.json')
    const sunday = DateTime.fromISO('2020-01-05')
    const events = await getEventsForWeek('hki', sunday)
    expect(events).toMatchSnapshot()
    nockDone()
  })
})

describe('parseCalendarHtml', () => {
  it('test', async () => {
    const html = fs.readFileSync(__dirname + '/fixtures/calendar.html', 'utf8')
    const datetimes = parseCalendarHtml(html)
    expect(
      sortBy((dt) => dt.toMillis(), datetimes).map((dt) => dt.toISO())
    ).toMatchSnapshot()
  })
})

describe('parseCreateBookingHref', () => {
  it('parses create booking link to datetime', async () => {
    const href =
      '/booking/create-booking?id_paketti=5&hlo=33&pvm=2020-01-05&klo=18%3A30%3A00'
    const datetime = parseCreateBookingHref(href)
    expect(datetime.isValid).toBe(true)
    expect(datetime.toISO()).toBe('2020-01-05T18:30:00.000+02:00')
  })
})
