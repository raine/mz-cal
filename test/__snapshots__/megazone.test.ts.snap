// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getEventsForWeek returns events with player counts 1`] = `
Array [
  Object {
    "end": "2019-12-31T12:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 16,
    "start": "2019-12-31T12:00:00.000+02:00",
  },
  Object {
    "end": "2020-01-02T12:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2020-01-02T12:00:00.000+02:00",
  },
  Object {
    "end": "2020-01-03T12:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 10,
    "start": "2020-01-03T12:00:00.000+02:00",
  },
  Object {
    "end": "2020-01-05T12:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 6,
    "start": "2020-01-05T12:00:00.000+02:00",
  },
  Object {
    "end": "2019-12-31T13:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 16,
    "start": "2019-12-31T12:30:00.000+02:00",
  },
  Object {
    "end": "2020-01-02T13:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2020-01-02T12:30:00.000+02:00",
  },
  Object {
    "end": "2020-01-03T13:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 8,
    "start": "2020-01-03T12:30:00.000+02:00",
  },
  Object {
    "end": "2020-01-05T13:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 6,
    "start": "2020-01-05T12:30:00.000+02:00",
  },
  Object {
    "end": "2019-12-31T13:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 28,
    "start": "2019-12-31T13:00:00.000+02:00",
  },
  Object {
    "end": "2020-01-02T13:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2020-01-02T13:00:00.000+02:00",
  },
  Object {
    "end": "2020-01-03T13:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2020-01-03T13:00:00.000+02:00",
  },
  Object {
    "end": "2020-01-04T13:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2020-01-04T13:00:00.000+02:00",
  },
  Object {
    "end": "2020-01-05T13:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 10,
    "start": "2020-01-05T13:00:00.000+02:00",
  },
  Object {
    "end": "2019-12-31T14:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 26,
    "start": "2019-12-31T13:30:00.000+02:00",
  },
  Object {
    "end": "2020-01-02T14:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2020-01-02T13:30:00.000+02:00",
  },
  Object {
    "end": "2020-01-03T14:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 2,
    "start": "2020-01-03T13:30:00.000+02:00",
  },
  Object {
    "end": "2020-01-04T14:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2020-01-04T13:30:00.000+02:00",
  },
  Object {
    "end": "2020-01-05T14:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 4,
    "start": "2020-01-05T13:30:00.000+02:00",
  },
  Object {
    "end": "2019-12-31T14:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2019-12-31T14:00:00.000+02:00",
  },
  Object {
    "end": "2020-01-02T14:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2020-01-02T14:00:00.000+02:00",
  },
  Object {
    "end": "2020-01-03T14:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 2,
    "start": "2020-01-03T14:00:00.000+02:00",
  },
  Object {
    "end": "2020-01-04T14:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2020-01-04T14:00:00.000+02:00",
  },
  Object {
    "end": "2020-01-05T14:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 7,
    "start": "2020-01-05T14:00:00.000+02:00",
  },
  Object {
    "end": "2019-12-31T15:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 2,
    "start": "2019-12-31T14:30:00.000+02:00",
  },
  Object {
    "end": "2020-01-02T15:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2020-01-02T14:30:00.000+02:00",
  },
  Object {
    "end": "2020-01-03T15:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2020-01-03T14:30:00.000+02:00",
  },
  Object {
    "end": "2020-01-04T15:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2020-01-04T14:30:00.000+02:00",
  },
  Object {
    "end": "2020-01-05T15:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 7,
    "start": "2020-01-05T14:30:00.000+02:00",
  },
  Object {
    "end": "2019-12-31T15:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 14,
    "start": "2019-12-31T15:00:00.000+02:00",
  },
  Object {
    "end": "2020-01-02T15:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 3,
    "start": "2020-01-02T15:00:00.000+02:00",
  },
  Object {
    "end": "2020-01-03T15:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2020-01-03T15:00:00.000+02:00",
  },
  Object {
    "end": "2020-01-04T15:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 31,
    "start": "2020-01-04T15:00:00.000+02:00",
  },
  Object {
    "end": "2020-01-05T15:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2020-01-05T15:00:00.000+02:00",
  },
  Object {
    "end": "2019-12-31T16:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 12,
    "start": "2019-12-31T15:30:00.000+02:00",
  },
  Object {
    "end": "2020-01-02T16:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2020-01-02T15:30:00.000+02:00",
  },
  Object {
    "end": "2020-01-03T16:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2020-01-03T15:30:00.000+02:00",
  },
  Object {
    "end": "2020-01-04T16:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 31,
    "start": "2020-01-04T15:30:00.000+02:00",
  },
  Object {
    "end": "2020-01-05T16:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2020-01-05T15:30:00.000+02:00",
  },
  Object {
    "end": "2019-12-31T16:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2019-12-31T16:00:00.000+02:00",
  },
  Object {
    "end": "2020-01-02T16:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2020-01-02T16:00:00.000+02:00",
  },
  Object {
    "end": "2020-01-03T16:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 11,
    "start": "2020-01-03T16:00:00.000+02:00",
  },
  Object {
    "end": "2020-01-04T16:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 14,
    "start": "2020-01-04T16:00:00.000+02:00",
  },
  Object {
    "end": "2020-01-05T16:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2020-01-05T16:00:00.000+02:00",
  },
  Object {
    "end": "2019-12-31T17:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 2,
    "start": "2019-12-31T16:30:00.000+02:00",
  },
  Object {
    "end": "2020-01-02T17:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2020-01-02T16:30:00.000+02:00",
  },
  Object {
    "end": "2020-01-03T17:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 11,
    "start": "2020-01-03T16:30:00.000+02:00",
  },
  Object {
    "end": "2020-01-04T17:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 29,
    "start": "2020-01-04T16:30:00.000+02:00",
  },
  Object {
    "end": "2020-01-05T17:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2020-01-05T16:30:00.000+02:00",
  },
  Object {
    "end": "2019-12-31T17:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 2,
    "start": "2019-12-31T17:00:00.000+02:00",
  },
  Object {
    "end": "2020-01-02T17:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2020-01-02T17:00:00.000+02:00",
  },
  Object {
    "end": "2020-01-03T17:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2020-01-03T17:00:00.000+02:00",
  },
  Object {
    "end": "2020-01-04T17:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 30,
    "start": "2020-01-04T17:00:00.000+02:00",
  },
  Object {
    "end": "2020-01-05T17:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2020-01-05T17:00:00.000+02:00",
  },
  Object {
    "end": "2019-12-31T18:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 4,
    "start": "2019-12-31T17:30:00.000+02:00",
  },
  Object {
    "end": "2020-01-02T18:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2020-01-02T17:30:00.000+02:00",
  },
  Object {
    "end": "2020-01-03T18:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 19,
    "start": "2020-01-03T17:30:00.000+02:00",
  },
  Object {
    "end": "2020-01-04T18:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 15,
    "start": "2020-01-04T17:30:00.000+02:00",
  },
  Object {
    "end": "2020-01-05T18:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2020-01-05T17:30:00.000+02:00",
  },
  Object {
    "end": "2019-12-31T18:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 4,
    "start": "2019-12-31T18:00:00.000+02:00",
  },
  Object {
    "end": "2020-01-02T18:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 8,
    "start": "2020-01-02T18:00:00.000+02:00",
  },
  Object {
    "end": "2020-01-03T18:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 19,
    "start": "2020-01-03T18:00:00.000+02:00",
  },
  Object {
    "end": "2020-01-04T18:30:00.000+02:00",
    "maxPlayers": 36,
    "players": 3,
    "start": "2020-01-04T18:00:00.000+02:00",
  },
  Object {
    "end": "2019-12-31T19:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2019-12-31T18:30:00.000+02:00",
  },
  Object {
    "end": "2020-01-02T19:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 8,
    "start": "2020-01-02T18:30:00.000+02:00",
  },
  Object {
    "end": "2020-01-03T19:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 0,
    "start": "2020-01-03T18:30:00.000+02:00",
  },
  Object {
    "end": "2020-01-04T19:00:00.000+02:00",
    "maxPlayers": 36,
    "players": 3,
    "start": "2020-01-04T18:30:00.000+02:00",
  },
]
`;

exports[`parseCalendarHtml test 1`] = `
Array [
  "2019-12-30T14:00:00.000+02:00",
  "2019-12-30T15:30:00.000+02:00",
  "2019-12-30T16:00:00.000+02:00",
  "2019-12-30T16:30:00.000+02:00",
  "2019-12-31T12:00:00.000+02:00",
  "2019-12-31T12:30:00.000+02:00",
  "2019-12-31T13:00:00.000+02:00",
  "2019-12-31T13:30:00.000+02:00",
  "2019-12-31T15:00:00.000+02:00",
  "2019-12-31T15:30:00.000+02:00",
  "2019-12-31T16:00:00.000+02:00",
  "2020-01-01T12:00:00.000+02:00",
  "2020-01-01T12:30:00.000+02:00",
  "2020-01-01T13:00:00.000+02:00",
  "2020-01-01T13:30:00.000+02:00",
  "2020-01-01T14:00:00.000+02:00",
  "2020-01-01T14:30:00.000+02:00",
  "2020-01-01T15:00:00.000+02:00",
  "2020-01-01T15:30:00.000+02:00",
  "2020-01-01T16:00:00.000+02:00",
  "2020-01-01T16:30:00.000+02:00",
  "2020-01-01T17:00:00.000+02:00",
  "2020-01-01T17:30:00.000+02:00",
  "2020-01-01T18:00:00.000+02:00",
  "2020-01-01T18:30:00.000+02:00",
  "2020-01-02T12:00:00.000+02:00",
  "2020-01-02T12:30:00.000+02:00",
  "2020-01-02T13:00:00.000+02:00",
  "2020-01-02T13:30:00.000+02:00",
  "2020-01-02T14:00:00.000+02:00",
  "2020-01-02T14:30:00.000+02:00",
  "2020-01-02T16:00:00.000+02:00",
  "2020-01-02T18:00:00.000+02:00",
  "2020-01-02T18:30:00.000+02:00",
  "2020-01-03T13:00:00.000+02:00",
  "2020-01-03T13:30:00.000+02:00",
  "2020-01-03T14:00:00.000+02:00",
  "2020-01-03T15:30:00.000+02:00",
  "2020-01-03T16:30:00.000+02:00",
  "2020-01-03T17:00:00.000+02:00",
  "2020-01-03T17:30:00.000+02:00",
  "2020-01-03T18:00:00.000+02:00",
  "2020-01-03T18:30:00.000+02:00",
  "2020-01-04T12:00:00.000+02:00",
  "2020-01-04T12:30:00.000+02:00",
  "2020-01-04T16:00:00.000+02:00",
  "2020-01-04T17:30:00.000+02:00",
  "2020-01-04T18:00:00.000+02:00",
  "2020-01-04T18:30:00.000+02:00",
  "2020-01-05T12:00:00.000+02:00",
  "2020-01-05T12:30:00.000+02:00",
  "2020-01-05T15:00:00.000+02:00",
  "2020-01-05T16:30:00.000+02:00",
  "2020-01-05T17:00:00.000+02:00",
  "2020-01-05T17:30:00.000+02:00",
  "2020-01-05T18:00:00.000+02:00",
  "2020-01-05T18:30:00.000+02:00",
]
`;
